app/src/main/java/com/example/todolistproject/view/TaskActivity.kt

```diff
@@ -0,0 +1,243 @@
+        // new object with the old one.
+        val modifiedTask = task.copy()
+        modifiedTask.title = dialogViewBinding.taskTitleEditText.text.toString()
+        modifiedTask.description = dialogViewBinding.taskDescriptionEditText.text.toString()
+        modifiedTask.startDate = dialogViewBinding.startDateEditText.text.toString()
+        modifiedTask.endDate = dialogViewBinding.endDateEditText.text.toString()
+
+        mViewModel.updateTask(modifiedTask)
+    }
+
+    private fun openDeleteTaskDialog(task: Task) {
+        val dialog = AlertDialog.Builder(this).setTitle(getString(R.string.todo_delete_task))
+            .setMessage(getString(R.string.todo_delete_task_confirmation))
+            .setPositiveButton(getString(R.string.todo_yes)) { _, _ ->
+                mViewModel.deleteTask(task)
+            }.setNegativeButton(getString(R.string.todo_no), null).create()
+        dialog.show()
+    }
+
+    private fun setTaskCompleted(task: Task) {
+        val isCompleted = task.isCompleted == true
+        task.isCompleted = !isCompleted
+
+        mViewModel.updateTask(task)
+    }
+
+    private fun openDatePickerDialog(editText: TextInputEditText) {
+        val calendar = Calendar.getInstance()
+        val dateSetListener = DatePickerDialog.OnDateSetListener { _, year, month, day ->
+            val date = Calendar.getInstance()
+            date.set(Calendar.YEAR, year)
+            date.set(Calendar.MONTH, month)
+            date.set(Calendar.DAY_OF_YEAR, day)
+            setDateText(editText, date)
+        }
+
+        DatePickerDialog(
+            this,
+            dateSetListener,
+            calendar.get(Calendar.YEAR),
+            calendar.get(Calendar.MONTH),
+            calendar.get(Calendar.DAY_OF_MONTH)
+        ).show()
+    }
+
+    private fun setDateText(editText: TextInputEditText, date: Calendar) {
+        val sdf = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault())
+        editText.setText(sdf.format(date.time))
+    }
+
+    fun openListFilter() {
+        val items = arrayOf(TASK, INCOMPLETE, COMPLETED)
+        val checkedItem = items.indexOf(currentFilter)
+        var tempFilter = currentFilter
+        MaterialAlertDialogBuilder(this)
+            .setTitle("Filter")
+            .setNegativeButton("Cancel") { dialog, _ ->
+                dialog.dismiss()
+            }
+            .setPositiveButton("OK") { dialog, _ ->
+                currentFilter = tempFilter
+                showTaskList(false)
+                setFilteredList()
+                dialog.dismiss()
+            }
+            .setSingleChoiceItems(items, checkedItem) { _, index ->
+                tempFilter = items[index]
+            }
+            .show()
+    }
+
+    private fun setFilteredList() {
+        when (currentFilter) {
+            TASK -> {
+                mViewModel.getTaskList().observe(this, {
+                    mAdapter.setTaskItems(it)
+                })
```

Comment:
```suggestion
                mViewModel.getTaskList().observe(this) {
                    mAdapter.setTaskItems(it)
                }
```
이런 표현 방법을 사용할수도 있어요(소소한이야기)